File: AndroidApp/app/src/main/java/com/malqaa/androidappp/newPhase/presentation/activities/productDetailsActivity/ImageViewLargeActivity.java
package com.malqaa.androidappp.newPhase.presentation.activities.productDetailsActivity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.RelativeLayout;

import com.malqaa.androidappp.R;
import com.malqaa.androidappp.newPhase.domain.models.ImageSelectModel;

import java.util.ArrayList;

public class ImageViewLargeActivity extends AppCompatActivity {
    ArrayList<String> mImageIds;



    private ViewPager mViewPager;
    ConstraintLayout layoutCancel;
    ImageView imageCancel;
    private ImageButton mBtnPrevious, mBtnNext;
    private ImagePagerAdapter mAdapter;
    private int mCurrentPosition = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.slider_display);

        ArrayList<ImageSelectModel> receivedList = getIntent().getParcelableArrayListExtra("imgList");




        imageCancel= findViewById(R.id.imageCancel);
        mViewPager = findViewById(R.id.viewPager);
        mBtnPrevious = findViewById(R.id.btnPrevious);
        layoutCancel = findViewById(R.id.layoutImg);
        mBtnNext = findViewById(R.id.btnNext);

        mAdapter = new ImagePagerAdapter(this, receivedList);
        mViewPager.setAdapter(mAdapter);

        imageCancel.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                setFinishOnTouchOutside(true);
                finish();
            }
        });

        layoutCancel.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                setFinishOnTouchOutside(true);
                finish();
            }
        });
        mBtnPrevious.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (mCurrentPosition > 0) {
                    mCurrentPosition--;
                    mViewPager.setCurrentItem(mCurrentPosition);
                }
            }
        });

        mBtnNext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (mCurrentPosition < mAdapter.getCount() - 1) {
                    mCurrentPosition++;
                    mViewPager.setCurrentItem(mCurrentPosition);
                }
            }
        });
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        finish();
    }
}

File: AndroidApp/app/src/main/java/com/malqaa/androidappp/newPhase/presentation/activities/productDetailsActivity/ImageViewLargeActivity.kt
package com.malqaa.androidappp.newPhase.presentation.activities.productDetailsActivity

import android.os.Build
import android.os.Bundle
import android.widget.ImageButton
import android.widget.ImageView
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.viewpager.widget.ViewPager
import com.malqaa.androidappp.R
import com.malqaa.androidappp.newPhase.domain.models.ImageSelectModel

class ImageViewLargeActivity : AppCompatActivity() {

    private lateinit var mViewPager: ViewPager
    private lateinit var layoutCancel: ConstraintLayout
    private lateinit var imageCancel: ImageView
    private lateinit var mBtnPrevious: ImageButton
    private lateinit var mBtnNext: ImageButton
    private lateinit var mAdapter: ImagePagerAdapter
    private var mCurrentPosition = 0
    private var UrlImg: String = ""

    @RequiresApi(Build.VERSION_CODES.N)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.slider_display)

        val receivedList: ArrayList<ImageSelectModel> =
            intent.getParcelableArrayListExtra("imgList") ?: ArrayList()

        UrlImg = (intent.getStringExtra("UrlImg") ?: "")
        receivedList.add(
            0, ImageSelectModel(
                null,
                "",
                false,
                UrlImg,
                false,
                0,
                1,
                false
            )
        )

        val distinctItems = receivedList.distinctBy { it.url }

        imageCancel = findViewById(R.id.imageCancel)
        mViewPager = findViewById(R.id.viewPager)
        mBtnPrevious = findViewById(R.id.btnPrevious)
        layoutCancel = findViewById(R.id.layoutImg)
        mBtnNext = findViewById(R.id.btnNext)

        mAdapter = ImagePagerAdapter(this, distinctItems)
        mViewPager.adapter = mAdapter
        mViewPager.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
            override fun onPageScrolled(
                position: Int,
                positionOffset: Float,
                positionOffsetPixels: Int
            ) {
                // Not used
            }

            override fun onPageSelected(position: Int) {
                // Release the ExoPlayer when the page is changed
                mAdapter.releaseStop()
            }

            override fun onPageScrollStateChanged(state: Int) {
                // Not used
            }
        })

        imageCancel.setOnClickListener {
            setFinishOnTouchOutside(true)
            finish()
        }

        layoutCancel.setOnClickListener {
            setFinishOnTouchOutside(true)
            finish()
        }

        mBtnPrevious.setOnClickListener {
            if (mCurrentPosition > 0) {
                mCurrentPosition--
                mViewPager.currentItem = mCurrentPosition
            }
        }

        mBtnNext.setOnClickListener {
            if (mCurrentPosition < mAdapter.count - 1) {
                mCurrentPosition++
                mViewPager.currentItem = mCurrentPosition
            }
        }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        finish()
    }

    override fun onPause() {
        super.onPause()
        mAdapter.releasePlayer()
    }

    override fun onStop() {
        super.onStop()
        mAdapter.releasePlayer()
    }
}
